### PR TITLE
Lfg 171: Splash Screen

### PR DESCRIPTION
**Pull request by kusalshrestha**

### CODE REVIEW

File: app/reducers/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from 'redux';
+import auth from './auth';
+
+const rootReducer = combineReducers({
+    auth
+});
+
+export default rootReducer;
```
Comment (leapfrog-saugat) [inline app/reducers/index.js]: add one line at EOF

File: app/screens/contact/contact.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    return (
+      <View style={ style.mainContainer }>
+        
+      </View>
+    );
+  }
+
+ }
+
+ export default ContactScreen
```
Comment (leapfrog-saugat) [inline app/screens/contact/contact.js]: semicolon at end of statement

Comment (leapfrog-saugat) [reply app/screens/contact/contact.js]: Please follow coding standards from https://lftechnology.atlassian.net/secure/RapidBoard.jspa?rapidView=306&view=detail&selectedIssue=BWA-388&quickFilter=1018

File: app/screens/login/login.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
+import React, { Component } from 'react';
+import { 
+  View, 
+  Text,
+  Image,  
+  TextInput,
+ } from 'react-native';
+
+import {Keyboard} from 'react-native'
```
Comment (leapfrog-saugat) [inline app/screens/login/login.js]: space after { and before }

File: app/screens/login/login.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                onSubmitEditing={ () => {
+                  this._validateEmail(this.state.email);                  
+                  this.passwordField.focus()
+                }}
+                value={this.state.email}
+              />
+            </View>
+            <View style={style.passwordFieldWrapper}>
+              <TextInput
+                ref={(component) => this.passwordField = component }
```
Comment (leapfrog-saugat) [inline app/screens/login/login.js]: follow js coding guidelines from the link mentioned above

File: app/constants/credentials.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export const IOS_GOOGLE_CLIENT_ID = '663889381642-iqslrisaiqqrr5dhbmmol8ahtos97v2d.apps.googleusercontent.com';
```
Comment (leapfrog-saugat) [inline app/constants/credentials.js]: don't include credentials and configs on git

File: app/reducers/index.js (hunk 1 above)
Comment (kusalshrestha) [reply app/reducers/index.js]: 🆗

File: app/screens/contact/contact.js (hunk 2 above)
Comment (kusalshrestha) [reply app/screens/contact/contact.js]: 🆗

File: app/screens/login/login.js (hunk 3 above)
Comment (kusalshrestha) [reply app/screens/login/login.js]: 🆗

File: app/screens/login/login.js (hunk 4 above)
Comment (kusalshrestha) [reply app/screens/login/login.js]: 🆗

File: app/constants/credentials.js (hunk 5 above)
Comment (kusalshrestha) [reply app/constants/credentials.js]: 🆗

File: app/screens/login/login.js [hunk 6]
```diff
@@ -22,14 +25,36 @@ import splash from '../../../assets/images/splash-screen.png';
...
 
   _login = () => {
-    console.log('mock login');
+    Keyboard.dismiss();
+    this._setErrorMessage();    
+    if (this.state.email === loginCredentials.email && this.state.password === loginCredentials.password) {
+      startTabScreen();      
+    } else {
+      // incorrect email / password
+      this._setErrorMessage('Incorrect Credentials!');
```
Comment (nishanbajracharya) [inline app/screens/login/login.js]: We should put these error messages in a separate file, `constants/messages` or `constants/errorMessages`.

File: app/screens/login/login.js [hunk 7]
```diff
@@ -62,44 +87,65 @@ import splash from '../../../assets/images/splash-screen.png';
     .done();
   }
 
+  _validateEmail = (email) => {
+    isEmailValid = validateEmail(email);
+    if (!isEmailValid) {
+      this._setErrorMessage('Invalid Email!');
```
Comment (nishanbajracharya) [inline app/screens/login/login.js]: Same here

**Review by nishanbajracharya — CHANGES_REQUESTED** [review]

**Review by nishanbajracharya — APPROVED** [review]

**Pull request merged**